x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java

```diff
@@ -254,6 +265,35 @@ public void updateExpirationTime(String docId,
         clientWithOrigin.update(request, listener);
     }
 
+    /**
+     * Attempts to update the stored response with failure
+     * This is called after we failed to update the stored response (e.g. because it exceeded allowed memory size),
+     * and we need to update the stored response with this failure.
+     * @param docId – document ID of the stored response
+     * @param responseHeaders – response headers
+     * @param response – response to store that contains the description of failure
+     */
+    public void updateResponseWithFailure(String docId,
```

Comment:
I think we don't need to disable the circuit breaker as a failure response should be small. If so, can we use `updateResponse` instead of this method?

Reply:
@dnhatn Thanks for additional comments. I've addressed this in ecf614d35339fe9f3388c6f82ca0d79d089a51ab